### DISCUSSION TITLE
LSP code action: remove all unused imported types and values

### DISCUSSION THREAD

**Discussion by nicklimmm** [gleam-lang/gleam#3241] [Ideas & suggestions]
I often find myself removing the unused imported types and values manually after I'm done with prototyping or refactoring.

Gleam can remove all unused imports, but not imported types and values yet.

A simple example:

```gleam
import gleam/list.{find} // ->`find` is unused
import gleam/uri.{type Uri} // -> `Uri` is unused
```

Should become:

```gleam
import gleam/list
import gleam/uri
```

This would be a good QOL improvement, especially if we are dealing with a lot of imported types and values.

**Comment by lpil** [comment]
Good idea! I've made an issue https://github.com/gleam-lang/gleam/issues/3257
